File: app/src/main/java/fr/corenting/edcompanion/models/apis/EDApi/Station.java
package fr.corenting.edcompanion.models.apis.EDApi;

import com.google.gson.annotations.SerializedName;

import java.util.Date;

public class Station {
    @SerializedName("distance_to_star")
    public int DistanceToStar;

    @SerializedName("last_shipyard_update")
    public Date LastShipyardUpdate;

    @SerializedName("max_landing_pad")
    public String MaxLandingPad;

    @SerializedName("name")
    public String Name;

    @SerializedName("type")
    public String Type;

    @SerializedName("system")
    public System System;
}

File: app/src/main/java/fr/corenting/edcompanion/models/Station.kt
package fr.corenting.edcompanion.models

import fr.corenting.edcompanion.models.apis.EDAPIV4.StationResponse
import org.threeten.bp.DateTimeUtils
import org.threeten.bp.Instant

data class Station(val name: String, val lastShipyardUpdate: Instant?,
                   val distanceToStar: Float, val maxLandingPad: String?, val isPlanetary: Boolean,
                   val type: String, val systemName: String) {

    companion object {
        fun fromStationResponse(res: StationResponse): Station {
            var lastShipyardUpdate: Instant? = null
            if (res.LastShipyardUpdate != null) {
                lastShipyardUpdate =  DateTimeUtils.toInstant(res.LastShipyardUpdate)
            }
            return Station(res.Name, lastShipyardUpdate,
                    res.DistanceToArrival, res.MaxLandingPad, res.IsPlanetary || res.IsSettlement, res.Type,
                    res.SystemName
            )
        }

        //fun fromCommodityDetailsStationResponse(res: CommodityDetailsResponse.CommodityDetailsStationResponse.CommodityDetailStationDetailsResponse): Station {
        //    var lastShipyardUpdate: Instant? = null
        //    if (res.LastShipyardUpdate != null) {
        //        lastShipyardUpdate =  DateTimeUtils.toInstant(res.LastShipyardUpdate)
        //    }
        //    return Station(res.Name, lastShipyardUpdate,
        //        res.DistanceToStar.toFloat(), res.MaxLandingPad, res.IsPlanetary, res.Type,
        //        res.System.Name
        //    )
        //}
    }
}